src/wallet/sqlite.cpp

```diff
@@ -144,7 +150,7 @@ SQLiteDatabase::~SQLiteDatabase()
     Cleanup();
 }
 
-void SQLiteDatabase::Cleanup() noexcept
+void SQLiteDatabase::Cleanup() noexcept EXCLUSIVE_LOCKS_REQUIRED(!g_sqlite_mutex)
```

Comment:
from developer-notes: "In functions that are declared separately from where they are defined, 
the thread safety annotations should be added exclusively to the function declaration."

Could move `g_sqlite_mutex` and `g_sqlite_count` to be static members of SQLiteDatabase` though I think.

Reply:
Moved to `SQLiteDatabase`. This way if some other method calls `Cleanup()` from the `.h` file they will be enforced to not hold the mutex. This is impossible though, because the mutex was defined solely in the `.cpp` file. I think either way is fine.

Unrelated, the developer-notes read:

> ... should be added exclusively to the function
    declaration. Annotations on the definition could lead to false positives
    (lack of compile failure) at call sites between the two.

Should that not be ["false negatives"](https://en.wikipedia.org/wiki/False_positives_and_false_negatives#False_negative_error) - the checks would wrongly assume that a caller does not hold the mutex?

What is "the two"?